docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,140 @@
+# North-South Messaging
+
+## Status
+
+** Proposed **
+(as of 12/21/21)
+
+## Context and Proposed Design
+
+Today, data flowing from sensors/devices (the “southside”) through EdgeX to enterprise applications, databases and cloud-based systems (the “northside”) can be accomplished via REST or Message bus.  That is, sensor or device data collected by a device service can be sent via REST or message bus to core data.  Core data then relays the data to application services via message bus, but the sensor data can also be sent directly from device services to application services via message bus (by passing core data).  The message bus is implemented via Redis Pub/Sub (default) or via MQTT.  From the application services, data can be sent to northside endpoints in any number of ways – including via MQTT.
+
+So, in summary, data can be collected from a sensor or device and be sent from the southside to the northside entirely using message bus technology when desired.
+
+!!! Note
+    The message bus may also be implemented in Zero MQ but with some limitations.  As Zero MQ is being deprecated in EdgeX, it will not be considered further in this ADR.
+
+Communications from a 3rd party system (enterprise application, cloud application, etc.) to EdgeX in order to acuate a device or get the latest information from a sensor is accomplished via REST.  The 3rd party system makes a REST call of the command service which then relays a request to a device service also using REST.  There is no built in means to make a message-based request of EdgeX or the devices/sensors it manages.
+
+In a future release of EdgeX, there is a desire to allow 3rd party systems to make requests of the southside via message bus.  Specifically, a 3rd party system will send a command request via message to the command service via the allowed message bus implementations (which could be MQTT or Redis Pub/Sub today).  The command service would then relay the message request via message bus to the managing device service.  (The command service would also perform any translations on the request as it does for REST requests today.) The device service would use the message to trigger action on the device/sensor as it does when it receives a REST request today and respond via message bus back to the command service.  In turn, the command service would respond to the 3rd party system via message bus.
+
+### Design
+
+The command service requires the means to subscribe to the EdgeX message bus.  It would use the messaging client (go-mod-messaging) to create a new MessageClient, connect to the message bus, and subscribe to a configured message topic.
```

Comment:
I think it could be either.  An adopter may choose to use their own message bus or use the EdgeX provided one - depending on whether they want to originate requests through their bus or ours.  I changed the document to reflect it could be either and should be configurable.